### DISCUSSION TITLE
Sequelize ORDER BY - Sort strings with number inside it

### DISCUSSION THREAD

**Discussion by elazsyahranie** [sequelize/sequelize#15529] [Help & Questions] [answered]
For example, I have a column named `SomeColumn` list of data like 

`DATA11`
`DATA12`
`DATA1`
`DATA2`
`DATA3`

I want to order them alphabetically. Obviously I should have added in my Sequelize like: 
`const getSomeData = await SomeModel.findAll({
    order: ["SomeColumn", "ASC"]
})`  

It works, but not 100% as I expected. `DATA11` and `DATA12` is sorted after `DATA1` and before `DATA2`. I searched the Internet and found a `StackExchange` forum here: `https://dba.stackexchange.com/questions/285222/how-to-treat-numbers-within-strings-as-numbers-when-sorting-a3-sorts-before`. 

It turns out I have to `CREATE COLLATIONS` in the Table (or the Database? I'm not sure, maybe you can correct me on this). So I have to: 

`CREATE COLLATION numeric (provider = icu, locale = 'en-u-kn-true');` 

And on the query, I have to `SELECT * FROM SomeTable ORDER BY SomeColumn COLLATE "numeric" ASC`. 

It does work. But the question is, how do I apply it on `Sequelize`? I prefer using methods other than `Sequelize literals`.

**Comment by ephys** [comment, marked as answer]
> I prefer using methods other than Sequelize `literals`. 

Literals are the only way to do this (currently):

```ts
-- in a migration or somewhere in your application, such as just after initializing sequelize:

await sequelize.query(`CREATE COLLATION IF NOT EXISTS numeric (provider = icu, locale = 'en-u-kn-true')`);

// When you query:

await User.findAll({
  order: [
    literal('SomeColumn COLLATE "numeric" ASC'),
  ],
});
```
